### ISSUE TITLE
Bus Error on Loading Alignment

### ISSUE THREAD

**Issue by A-Swafford** [veg/hyphy#688]
I am attempting to load an alignment into HYPHY (v. 2.3.420171102beta(MP)) and get either a Segmentation Fault (core dumped) or a Bus Error (core dumped) when loading a specific alignment.

I'm working on Ubuntu 16.04 and have ~23GB of memory available.  The alignment is in fasta format (are there restrictions on characters that can appear in the headers?), here is a small example:

https://pastebin.com/83iEZS8c

I have also reduced the headers to small, unique strings with only alpha-numeric characters and continue to get the same error.

**Comment by spond** [comment]
Dear @A-Swafford,

Could you please specify how you are trying to load the alignment? Simply reading it in, via something like

```
DataSet test = ReadDataFile ("issue688.fna");
DataSetFilter filter = CreateFilter (test, 1);
```

does not seem to create issues with the current release (2.3.6); the file is read and the program exits normally.

Best,
Sergei

**Comment by A-Swafford** [comment]
@spond,

I have not tried using script to load it, this was simply using the command-line prompts that are given once you run HYPHY.

**Comment by spond** [comment]
Dear @A-swafford,

Could you provide the complete terminal session transcript that leads to the bus error? A segfault is not acceptable in any situation. 

Best,
Sergei.

**Comment by A-Swafford** [comment]
Certainly!

[Hyphy_SegFault_Test.txt](https://github.com/veg/hyphy/files/1439752/Hyphy_SegFault_Test.txt)

I've also attached the "test.aln" file (renamed as "test.txt" for github), which is a severely reduced and simplified version of the full alignment.
[test.txt](https://github.com/veg/hyphy/files/1439755/test.txt)

This could be an issue with my Ubuntu install, though I did a fresh install alongside windows this morning before posting here, please let me know if there's any troubleshooting/information I can do on this end. I am relatively new to Linux and may have overlooked some simple prereq. that is not automatically downloaded with apt-get update.

**Comment by stephenshank** [comment]
Dear @A-Swafford ,

I tried to run MEME on your data and can indeed replicate the issue. However, your alignment seems to be comprised of amino acids, and I get a similar error whenever I run MEME on _any_ amino acid alignment. I believe the issue here is that MEME, being a codon-based method, will only accept nucleotide/codon alignments that were back-translated from corresponding protein alignments (using a tool like pal2nal).

So my guess is that a nice feature for HyPhy would be to give a more descriptive error message in this scenario, rather than a segmentation fault. Just some food for thought... @spond, please feel free to correct.

Best,
Stephen

**Comment by spond** [comment]
Dear @A-Swafford,

In addition to what @stephenshank pointed out, your example also exposed a different bug (that was causing the crash). Briefly, HyPhy was constructing a data set of amino-acid triplets (instead of nucleotide triplets), and incorrectly computing the number of possible characters (64) instead of the correct number of 8000. This was causing memory issues and led to the crash. This has been fixed with commit f3d2f03 to the beta branch. The program will now print an error message before terminating, since MEME indeed cannot be run on amino-acid data.

Thanks for reporting,
Sergei
